src/libdrakvuf/drakvuf.c

```diff
@@ -1085,6 +1085,75 @@ bool drakvuf_set_vcpu_gprs(drakvuf_t drakvuf, unsigned int vcpu, registers_t* re
     return xen_set_vcpu_ctx(drakvuf->xen, drakvuf->domID, vcpu, &ctx);
 }
 
+bool drakvuf_vmi_response_set_registers(
+    drakvuf_t drakvuf,
+    drakvuf_trap_info_t* info,
+    x86_registers_t* regs,
+    bool immediate)
+{
+    bool res;
+
+    drakvuf_lock_and_get_vmi(drakvuf);
+    if (drakvuf->vmi_response_set_registers[info->vcpu])
+    {
+        PRINT_DEBUG("[LIBDRAKVUF] [%ld] Failed to set registers\n", info->event_uid);
+        res = false;
+    }
+    else
+    {
+        drakvuf->vmi_response_set_registers[info->vcpu] = true;
+        memcpy(&info->regs_modified, regs, sizeof(x86_registers_t));
```

Comment:
Why keep the modified regs in the trap info structure? Other plugins and callbacks should not need access to it, so would make more sense to keep it in a libdrakvuf private lookup structure.

Reply:
I completely agree with you.

Though there have been issues with includes. This would require some more heavy refactoring. So I have decided to send current modifications as is. And make some refactoring as soon as possible.

Reply:
@tklengyel is this OK to you?

Reply:
I don't follow, why would there be a refactor needed? The regs are being passed in as input to this function, you can store them anywhere. Better to store them internally if its not intended to be consumable by multiple plugins.

Reply:
I have an idea. I try it.

Reply:
Fixed